ros2/content.md

```diff
@@ -0,0 +1,265 @@
+For example, if one wishes to use their own `.ros` folder that already resides in their local home directory, with a username of `ubuntu`, we can simple launch the container with an additional volume argument:
+
+```console
+$ docker run -v "/home/ubuntu/.ros/:/root/.ros/" %%IMAGE%%
+```
+
+### Devices
+
+Some application may require device access for acquiring images from connected cameras, control input from human interface device, or GPUS for hardware acceleration. This can be done using the [`--device`](https://docs.docker.com/reference/run/) run argument to mount the device inside the container, providing processes inside hardware access.
+
+### Networks
+
+ROS 2 allows for peer-to-peer networking of processes (potentially distributed across machines) that are loosely coupled using the ROS communication infrastructure. ROS implements several different styles of communication, including synchronous RPC-style communication over services, asynchronous streaming of typed data over topics, combinations of both prior via request/reply and status/feedback over actions, and run-time settings via configuration over parameters. To abide by the best practice of [one process per container](https://docs.docker.com/articles/dockerfile_best-practices/), Docker networks can be used to string together several running ROS processes. For further details see the Deployment example further below.
+
+Alternatively, more permissive network setting can be use to share all host network interfaces with the container, such as [`host` network driver](https://docs.docker.com/network/host/), simplifying connectivity with external network participants. Be aware however that this removes the networking namespace separation between containers, and can affect the ability of DDS participants communicate between containers, as documented [here](https://community.rti.com/kb/how-use-rti-connext-dds-communicate-across-docker-containers-using-host-driver).
+
+## Deployment example
+
+### Docker Compose
+
+In this example we'll demonstrate using [`docker-compose`](https://docs.docker.com/compose/) to spawn a pair of message publisher and subscriber nodes in seperate containers connected through shared software defined network.
+
+> Create the directory `~/ros2_demos` and add the first `Dockerfile` example from above. In the same directory, also create file `docker-compose.yml` with the following that runs a C++ publisher with a Python subscriber:
+
+```yaml
+version: '3'
+
+services:
+  talker:
+    build: ./Dockerfile
+    command: ros2 run demo_nodes_cpp talker
+
+  listener:
+    build: ./Dockerfile
+    command: ros2 run demo_nodes_py listener
+```
+
+> Use docker-copose inside the same directory to launch our ROS nodes. Given the containers created derive from the same docker compose project, they will coexist on shared project network:
+
+```console
+$ docker-compose up -d
+```
+
+> Notice that a new network named `ros2_demos` has been created, as can be shown further with:
```

Comment:
Is this little trailing intro to docker-compose worth keeping, or should we assume the basics of docker-compose is common knowledge?